highlight.js

```diff
@@ -538,3 +538,60 @@ function fuzzy(position, size, shape, text, service, callback) {
         rect(x, y, width, height, text, service, callback);
     }
 }
+
+function showImage(path, size, text, service, callback) {
+    // background overlay
+    const monitors = Main.layoutManager.monitors;
+    const primary = Main.layoutManager.primaryIndex;
+    const monitor = monitors[primary];
+    const [width, height] = fuzzyParser.parseSize(size);
+
+    let textActor = null;
+    let skipButton = null;
+
+    if (service.Skippable) {
+        skipButton = _createSkipButton(callback);
+        Actors.push(skipButton);
+    }
+
+    if (text) {
+        textActor = _createText(text);
+        Actors.push(textActor);
+    }
+
+    const cache = St.TextureCache.get_default();
+    const scaleFactor = St.ThemeContext.get_for_stage(global.stage).scale_factor;
+
+    const file = Gio.file_new_for_path(path);
+    const actor = cache.load_file_async(file, -1, -1, scaleFactor, 1);
+    actor.set_size(width, height);
+    const container = new St.BoxLayout({
+        style_class: 'osd-window',
+        vertical: true,
+    });
+    container.add_actor(actor);
+
+    actor.connect('notify::allocation', () => {
+        const [w, h] = container.get_size();
+        const x = monitor.x + monitor.width / 2 - w / 2;
+        const y = monitor.y + monitor.height / 2 - h / 2;
+
+        // This is from the osd-window margin: in
+        // gnome-shell:data/theme/gnome-shell-sass/widgets/_osd.scss
+        // > margin: $base_margin * 8; // 32px
+        const margin = 32;
```

Comment:
This could be a constant outside the function, instead of being assigned each time showImage is called.

Reply:
I've moved this to a global variable as suggested.